scipy/stats/_distribution_infrastructure.py

```diff
@@ -3417,6 +3567,117 @@ def plot(self, x='x', y='pdf', *, t=('cdf', 0.0005, 0.9995), ax=None):
     # these methods reasonably efficiently.
 
 
+class ContinuousDistribution(_UnivariateDistribution):
+    def _overrides(self, method_name):
+        if method_name in {'_logpmf_formula', '_pmf_formula'}:
+            return True
+        return super()._overrides(method_name)
+
+    def _pmf_formula(self, x, **params):
+        return np.zeros_like(x)
+
+    def _logpmf_formula(self, x, **params):
+        return np.full_like(x, -np.inf)
+
+    def _pxf_dispatch(self, x, *, method=None, **params):
+        return self._pdf_dispatch(x, method=method, **params)
+
+    def _logpxf_dispatch(self, x, *, method=None, **params):
+        return self._logpdf_dispatch(x, method=method, **params)
+
+
+class DiscreteDistribution(_UnivariateDistribution):
+    def _overrides(self, method_name):
+        if method_name in {'_logpdf_formula', '_pdf_formula'}:
+            return True
+        return super()._overrides(method_name)
+
+    # These should probably check whether pmf = 0
+    def _logpdf_formula(self, x, **params):
+        return np.full_like(x, np.inf)
+
+    def _pdf_formula(self, x, **params):
+        return np.full_like(x, np.inf)
+
+    def _pxf_dispatch(self, x, *, method=None, **params):
+        return self._pmf_dispatch(x, method=method, **params)
+
+    def _logpxf_dispatch(self, x, *, method=None, **params):
+        return self._logpmf_dispatch(x, method=method, **params)
+
+    def _cdf_quadrature(self, x, **params):
+        return super()._cdf_quadrature(np.floor(x), **params)
+
+    def _logcdf_quadrature(self, x, **params):
+        return super()._logcdf_quadrature(np.floor(x), **params)
+
+    def _ccdf_quadrature(self, x, **params):
+        return super()._ccdf_quadrature(np.floor(x + 1), **params)
+
+    def _logccdf_quadrature(self, x, **params):
+        return super()._logccdf_quadrature(np.floor(x + 1), **params)
+
+    def _icdf_inversion(self, x, **params):
+        res = self._solve_bounded(self._cdf_dispatch, x, params=params, xatol=1)
+        res = np.where(res.fl >= 0, np.floor(res.xl), np.floor(res.xr))
+        res[np.isnan(x)] = np.nan
+        return res
+
+    def _ilogcdf_inversion(self, x, **params):
+        res = self._solve_bounded(self._logcdf_dispatch, x, params=params, xatol=1)
+        res = np.where(res.fl >= 0, np.floor(res.xl), np.floor(res.xr))
+        res[np.isnan(x)] = np.nan
+        return res
+
+    def _iccdf_inversion(self, x, **params):
+        res = self._solve_bounded(self._ccdf_dispatch, x, params=params, xatol=1)
+        res = np.where(res.fl <= 0, np.floor(res.xl), np.floor(res.xr))
+        res[np.isnan(x)] = np.nan
+        return res
+
+    def _ilogccdf_inversion(self, x, **params):
+        res = self._solve_bounded(self._logccdf_dispatch, x, params=params, xatol=1)
+        res = np.where(res.fl <= 0, np.floor(res.xl), np.floor(res.xr))
+        res[np.isnan(x)] = np.nan
+        return res
```

Comment:
Here's the special case handling in change 3. We might be able to eliminate it if we handle the case in which there is no valid bracket.